Models/RecognitionViewModel.swift

```diff
@@ -0,0 +1,110 @@
+    private let historyFileURL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)[0].appendingPathComponent("history.json")
+    
+    init() {
+        loadHistory()
+    }
+    
+    func recognizeObject(in image: UIImage) {
+        guard let model = try? VNCoreMLModel(for: Resnet50(configuration: .init()).model) else {
+            errorMessage = "Nie zaladowano coreML"
+            return
+        }
+        
+        let request = VNCoreMLRequest(model: model) { [weak self] request, error in
+            guard let results = request.results as? [VNClassificationObservation],
+                  let topResult = results.first else {
+                self?.errorMessage = "Nie udalo sie rozpoznac"
+                return
+            }
+            
+            let resultText = "Objekt: \(topResult.identifier)\nPewność: \(Int(topResult.confidence * 100))%"
+            self?.recognitionResult = resultText
+            
+            let record = RecognitionRecord(
+                id: UUID(),
+                image: image,
+                result: resultText,
+                date: Date()
+            )
+            
+            DispatchQueue.main.async {
+                self?.history.insert(record, at: 0)
+                if self?.history.count ?? 0 > self?.maxHistoryCount ?? 10 {
+                    self?.history.removeLast()
+                }
+                self?.saveHistory()
+            }
+        }
+        
+        request.imageCropAndScaleOption = .scaleFit
+        
+        guard let cgImage = image.cgImage else {
+            errorMessage = "Zly obraz"
+            return
+        }
+        
+        let handler = VNImageRequestHandler(cgImage: cgImage, options: [:])
+        try? handler.perform([request])
+    }
+    
+    func saveHistory() {
+        let encoder = JSONEncoder()
+        encoder.dateEncodingStrategy = .iso8601
+        do {
+            let data = try encoder.encode(history.map { HistoryEntry(record: $0) })
+            try data.write(to: historyFileURL)
+        } catch {
+            print("Nie udalo sie zapiasc do historii: \(error)")
+        }
+    }
+    
+    func loadHistory() {
+        let decoder = JSONDecoder()
+        decoder.dateDecodingStrategy = .iso8601
+        do {
+            let data = try Data(contentsOf: historyFileURL)
+            let entries = try decoder.decode([HistoryEntry].self, from: data)
+            history = entries.compactMap { $0.toRecord() }
+        } catch {
+            print("Nie udalo sie zaladowac historii: \(error)")
+        }
+    }
+    
+    func exportResult(_ record: RecognitionRecord) {
+        let text = "Wynik analizy\nData: \(record.date.formatted())\n\(record.result)"
+        let activityController = UIActivityViewController(activityItems: [text, record.image], applicationActivities: nil)
+        UIApplication.shared.windows.first?.rootViewController?.present(activityController, animated: true)
```

Comment:
UIApplication.shared.windows.first jest przestarzałe
Lepiej użyć:
```suggestion
if let scene = UIApplication.shared.connectedScenes.first as? UIWindowScene,
   let root = scene.windows.first?.rootViewController {
    root.present(activityController, animated: true)
}
```